test/test_torch.py

```diff
@@ -17532,6 +17532,19 @@ def test_pickle_gradscaler(self, device):
                 if lazy_init_scale:
                     self.assertEqual(b.scale(torch.tensor([4.0], dtype=torch.float32, device=device)), 12.0)
 
+    @dtypesIfCUDA(torch.half, torch.float, torch.double,
+                  torch.int8, torch.short, torch.int, torch.long)
+    @dtypes(torch.float, torch.double,
+            torch.int8, torch.short, torch.int, torch.long)
+    def test_nansum(self, device, dtype):
+        x = (torch.randn(3, 3))
+        if dtype in [torch.half, torch.float, torch.double]:
```

Comment:
`if dtype.is_floating_point:`

But more generally we should avoid taking a list of a tensor. It's better to test for floating point or not and call the appropriate tensor construct on the device and with the given dtype. That will avoid creating a tensor to take a list of it to pass to compare_with_numpy to create yet another tensor.